File: python/src/calc_stats2.py
import statistics


def maximum(values: list[int]):
    # bug - should throw ValueError on empty list
    result = float('-inf')
    for value in values:
        if value > result:
            result = value
    return result


def minimum(values: list[int]):
    # bug - should throw ValueError on empty list
    result = values[0]
    for value in values:
        # bug - should be value < result
        if value > result:
            result = value
    return result


def count(values: list[int]):
    result = 0
    for v in values:
        result += 1
        # bug
        result += 1
    return result


def average(values: list[int]):
    # bug - should throw ValueError on empty list
    if len(values) < 1:
        return 0
    sum = 0
    for v in values:
        sum += v
    return sum / float(len(values))

File: php/src/CalcStats2.php
<?php

namespace CodingDojo;

class CalcStats2
{

    public static function minimum($numbersList)
    {
        // bug - should throw IllegalArgumentException on empty list
        $min = $numbersList[0];
        foreach ($numbersList as $n) {
            // bug, should be $n < $min
            if ($n > $min) {
                $min = $n;
            }
        }
        return $min;
    }

    public static function maximum($numbersList)
    {
        // bug - should throw IllegalArgumentException on empty list
        $max = PHP_INT_MIN;
        foreach ($numbersList as $n) {
            if ($n > $max) {
                $max = $n;
            } else {
                // bug, should not do this
                $max = PHP_INT_MIN;
            }
        }
        return $max;
    }

    public static function average($numbersList)
    {
        // bug - should throw IllegalArgumentException on empty list
        if (count($numbersList) < 1) {
            return 0;
        }
        // bug, should be 0
        $sum = 1;
        foreach ($numbersList as $n) {
            $sum += $n;
        }

        return $sum / count($numbersList);
    }

    public static function count($numbersList)
    {
        $count = 0;
        foreach ($numbersList as $n) {
            $count++;
            // bug, shouldn't do this twice
            $count++;
        }
        return $count;
    }
}
